subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c

```diff
@@ -531,37 +643,141 @@ static int isr_rx(struct lll_sync *lll, uint8_t node_type, uint8_t *crc_ok)
 	}
 #endif /* CONFIG_BT_CTLR_DF_SAMPLE_CTE_FOR_PDU_WITH_BAD_CRC */
 
-isr_rx_done:
 	return err;
 }
 
+static void isr_rx_adv_sync_estab(void *param)
+{
+	enum sync_status sync_ok;
+	struct lll_sync *lll;
+	uint8_t rssi_ready;
+	uint8_t trx_done;
+	uint8_t crc_ok;
+	int err;
+
+	lll = param;
+
+	/* Read radio status and events */
+	trx_done = radio_is_done();
+	if (trx_done) {
+		crc_ok = radio_crc_is_valid();
+		rssi_ready = radio_rssi_is_ready();
+		sync_ok = sync_filtrate_by_cte_type(lll->cte_type, lll->filter_policy);
+		++trx_cnt;
+	} else {
+		crc_ok = rssi_ready = 0U;
+		/* Initiated as allowed, crc_ok takes precended during handling of PDU
+		 * reception in the situation.
+		 */
+		sync_ok = SYNC_STAT_ALLOWED;
+	}
+
+	/* Clear radio rx status and events */
+	lll_isr_rx_status_reset();
+
+	/* No Rx */
+	if (!trx_done) {
+		/* TODO: Combine the early exit with above if-then-else block
+		 */
+		goto isr_rx_done;
+	}
+
+	/* Save radio ready and address capture timestamp for later use for
+	 * drift compensation.
+	 */
+	radio_tmr_aa_save(radio_tmr_aa_get());
+	radio_tmr_ready_save(radio_tmr_ready_get());
+
+	/* Handle regular PDU reception if CTE type is acceptable */
+	if (sync_ok == SYNC_STAT_ALLOWED) {
+		err = isr_rx(lll, NODE_RX_TYPE_SYNC, crc_ok, rssi_ready, SYNC_STAT_ALLOWED);
+		if (err == -EBUSY) {
+			return;
+		}
+	} else if (sync_ok == SYNC_STAT_TERM) {
+		struct node_rx_pdu *node_rx;
+
+		node_rx = ull_pdu_rx_alloc_peek(3);
+		if (node_rx) {
+			struct node_rx_ftr *ftr;
+
+			ull_pdu_rx_alloc();
```

Comment:
Can you elaborate on why we need `peek(3)` for a single alloc?

Reply:
`ull_pdu_rx_alloc_peek` name is a bit unfortunate. It peeks one buffer but only in case there are `x` buffers available. This is to ensure there are buffers for other use cases. 
In this case the verification is to ensure there are nodes for: reporting received PDU (the one that is allocated), 1x to receive a PDU in a connection, 1x to receive empty PDU.

I'll add a comment above.

BTW. I'm not going to change the function name ;-)